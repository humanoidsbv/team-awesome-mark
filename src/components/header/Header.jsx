import React, { useState } from 'react';
import * as Styled from './Header.styled';

export const Header = () => {
    const [isActive, setIsActive] = useState(false);
    const handleClick = () => {
        setIsActive(!isActive);
    }

    return (
    <Styled.Container isActive={isActive}>
        <Styled.Logo>team awesome</Styled.Logo>
            <Styled.Menu isActive={isActive}>
                <Styled.MenuItems>Timesheets</Styled.MenuItems>
                <Styled.MenuItems>Team members</Styled.MenuItems>
                <Styled.MenuItems>Projects</Styled.MenuItems>
                <Styled.MenuItems>Clients</Styled.MenuItems>
                <Styled.MenuItems>Documents</Styled.MenuItems>
            </Styled.Menu>
        <Styled.Toggle onClick={handleClick}>x</Styled.Toggle>
    </Styled.Container>
    );
}
