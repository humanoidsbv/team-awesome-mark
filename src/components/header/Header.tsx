import Link from "next/link";
import React, { useState } from "react";

import * as Styled from "./Header.styled";
import CloseIcon from "../../../public/images/close.svg";
import MenuIcon from "../../../public/images/menu.svg";

export const Header = () => {
  const [isActive, setIsActive] = useState(false);
  const handleClick = () => {
    setIsActive(!isActive);
  };

  return (
    <Styled.Container isActive={isActive}>
      <Styled.Logo>team awesome</Styled.Logo>
      <Styled.Menu isActive={!isActive}>
        <Link href="/">
          <Styled.MenuItems>Timesheets</Styled.MenuItems>
        </Link>
        <Link href="/team-members">
          <Styled.MenuItems>Team members</Styled.MenuItems>
        </Link>
        <Styled.MenuItems>Projects</Styled.MenuItems>
        <Styled.MenuItems>Clients</Styled.MenuItems>
        <Styled.MenuItems>Documents</Styled.MenuItems>
      </Styled.Menu>
      <Styled.Toggle onClick={handleClick}>
        {!isActive ? <MenuIcon /> : <CloseIcon />}
      </Styled.Toggle>
    </Styled.Container>
  );
};
