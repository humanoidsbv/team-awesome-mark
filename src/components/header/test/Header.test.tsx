/* eslint-disable import/no-extraneous-dependencies */
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import { Header } from "../Header";

test("if hamburger button is clicked", async () => {
  const header = render(<Header />);
  const button = header.getByRole("button");
  const hamburger = header.getByTestId("hamburger");

  expect(hamburger).toBeInTheDocument();

  fireEvent.click(button);

  const close = header.getByTestId("close");
  expect(close).toBeInTheDocument();
});
