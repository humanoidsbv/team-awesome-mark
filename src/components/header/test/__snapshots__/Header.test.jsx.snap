// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="Headerstyled__Container-sc-9t0655-0 gMEqcK"
>
  <a
    className="Headerstyled__Logo-sc-9t0655-2 kJdOvu"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    team awesome
  </a>
  <ul
    className="Headerstyled__Menu-sc-9t0655-3 fhrFOB"
  >
    <li
      className="Headerstyled__MenuItems-sc-9t0655-4 iEawjO"
    >
      <a
        className="Headerstyled__Item-sc-9t0655-1 cEhQft"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Timesheets
      </a>
    </li>
    <li
      className="Headerstyled__MenuItems-sc-9t0655-4 iEawjO"
    >
      <a
        className="Headerstyled__Item-sc-9t0655-1 cEhQft"
        data-cy="team-members-link"
        href="/team-members"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Team members
      </a>
    </li>
    <li
      className="Headerstyled__MenuItems-sc-9t0655-4 iEawjO"
    >
      <a
        className="Headerstyled__Item-sc-9t0655-1 cEhQft"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Projects
      </a>
    </li>
    <li
      className="Headerstyled__MenuItems-sc-9t0655-4 iEawjO"
    >
      <a
        className="Headerstyled__Item-sc-9t0655-1 cEhQft"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Clients
      </a>
    </li>
    <li
      className="Headerstyled__MenuItems-sc-9t0655-4 iEawjO"
    >
      <a
        className="Headerstyled__Item-sc-9t0655-1 cEhQft"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Documents
      </a>
    </li>
  </ul>
  <button
    className="Headerstyled__Toggle-sc-9t0655-5 Mmodw"
    onClick={[Function]}
  >
    <svg
      data-cy="hamburger-icon"
      data-testid="hamburger-icon"
      fill="none"
      height="12"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 12h16c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1Zm0-5h16c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1ZM0 1c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1Z"
        fill="#fff"
      />
    </svg>
  </button>
</nav>
`;
